deltacat/compute/compactor_v2/steps/merge.py

```diff
@@ -345,20 +466,44 @@ def _timed_merge(input: MergeInput) -> MergeResult:
         f"merge_{worker_id}_{task_id}.bin"
     ) if input.enable_profiler else nullcontext():
         total_input_records, total_deduped_records = 0, 0
+        total_dropped_records = 0
         materialized_results: List[MaterializeResult] = []
         merge_file_groups = input.merge_file_groups_provider.create()
         hb_index_copy_by_ref_ids = []
 
         for merge_file_group in merge_file_groups:
-            if not merge_file_group.dfe_groups:
+            compacted_table = None
+            has_delete = input.delete_file_envelopes is not None
```

Comment:
Have you considered adding delete envelopes to `MergeFileGroup` and updating the provider so that it doesn't have to be directly fed from the input object?